### PR TITLE
refactor(ui5-combobox): integrate properly popover api

### PR DESCRIPTION
**Pull request by nikoletavnv**
Adjustments in ui5-combobox after integrating the popover api. 
Async methods are replaced in favor of sync methods. 
Popover state is controlled by property.

### CODE REVIEW

**Review by ivoplashkov — CHANGES_REQUESTED** [review]
A couple issues I have found:
- focus the first input with value state and press Tab, value state popover does not get closed and next one gets opened.
- Focus the input with valueState=“Warning” press escape to close the popover and then tab out (styles remain as it is focused) and focus it again, it does not get opened anymore.

**Comment by nikoletavnv** [comment]
> A couple issues I have found:
> 
> * focus the first input with value state and press Tab, value state popover does not get closed and next one gets opened.
> * Focus the input with valueState=“Warning” press escape to close the popover and then tab out (styles remain as it is focused) and focus it again, it does not get opened anymore.

These 2 are fixed now.

**Review by ivoplashkov — CHANGES_REQUESTED** [review]
I have left some comments but I have realized that you don't have the latest changes from the main branch. The logic behind the group items is quite different now, could you please check and I will review again?

File: packages/main/src/ComboBox.ts [hunk 1]
```diff
@@ -534,15 +522,19 @@ class ComboBox extends UI5Element implements IFormInputElement {
...
-		if (!(this.getDomRef()!.contains(toBeFocused)) && (popover !== e.relatedTarget)) {
+		if (!(this.getDomRef()!.contains(toBeFocused)) && (this._getPicker() !== e.relatedTarget)) {
 			this.focused = false;
-			!isPhone() && this._closeRespPopover(e);
+		}
+	}
+
+	_beforeOpenPopover() {
+		if (isPhone()) {
+			this._getPicker()!.querySelector<HTMLInputElement>("[ui5-input]")!.value = this.value;
```
[the remark this answers is not in the document]
Comment (nikoletavnv) [reply packages/main/src/ComboBox.ts]: Done

File: packages/main/src/ComboBox.ts [hunk 2]
```diff
@@ -648,6 +614,12 @@ class ComboBox extends UI5Element implements IFormInputElement {
 		this._toggleRespPopover();
 	}
 
+	_handleMobileKeydown(e: KeyboardEvent) {
+		if (isEscape(e)) {
+			this._closeRespPopover(null, true);
```
[the remark this answers is not in the document]
Comment (nikoletavnv) [reply packages/main/src/ComboBox.ts]: Done

**Review by ivoplashkov — CHANGES_REQUESTED** [review]
1. The sample with valueState="Information" does not show valueState popover upon focusing the input.
2. The sample with valueState="Erorr" and link, if you click the link and then press Tab, focus get trapped. ( seems like the valueState could be focused upon click, which is incorrect, I think we need to move the fake focus to the valueState only on arrow navigation)

**Review by ivoplashkov — CHANGES_REQUESTED** [review]
As discussed:

1. Focus should go to the first real item in case of grouping.
2. When face focus goes to the ValueStateMessage and user presses Tab, focus should go to the link if any.
3. Write some tests for the common issues you have fixed.

**Comment by nikoletavnv** [comment]
> 1. The sample with valueState="Information" does not show valueState popover upon focusing the input.
> 2. The sample with valueState="Erorr" and link, if you click the link and then press Tab, focus get trapped. ( seems like the valueState could be focused upon click, which is incorrect, I think we need to move the fake focus to the valueState only on arrow navigation)

These issues are fixed now

**Comment by nikoletavnv** [comment]
> As discussed:
> 
> 1. Focus should go to the first real item in case of grouping.
> 2. When fake focus goes to the ValueStateMessage and user presses Tab,  real focus should go to the link if any.
> 3. Write some tests for the common issues you have fixed.

Done

File: packages/main/src/ComboBox.ts [hunk 3]
```diff
@@ -971,6 +964,22 @@ class ComboBox extends UI5Element implements IFormInputElement {
...
+					// Focus the first link inside the value state message if visual focus is on the message and the user presses tab
+					setTimeout(() => {
+						const firstLink = this._getPicker().querySelector("*[slot='header']")!.querySelector("*[slot='valueStateMessage']")!.querySelector("a")!;
+						firstLink.tabIndex = 0;
+						firstLink.focus();
+					}, 100);
+					return;
+				}
+			}
+
```
Comment (ivoplashkov) [inline packages/main/src/ComboBox.ts:982]: Could we please think of another approach for this use case as this seems to be very complex and unreliable. If you think that handling this case means spending a lot of time on research we may create another BLI for this case only ( I have discussed this with our PO ).

**Review by ivoplashkov — APPROVED** [review]

**Pull request merged**